sale_order_general_discount/models/sale.py

```diff
@@ -48,3 +48,18 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False,
                 order_line_field.attrib['context'] = context
                 res['arch'] = etree.tostring(order_xml)
         return res
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    @api.model
+    def create(self, vals):
+        """Apply general discount for sale order lines which are not created
+        from sale order form view.
+        """
+        if 'discount' not in vals and 'order_id' in vals:
+            sale_order = self.env['sale.order'].browse(vals['order_id'])
+            if sale_order.general_discount:
+                vals['discount'] = sale_order.general_discount
```

Comment:
LGTM but could be done also with an default value on `discount` from `order_id.general_discount`?

Reply:
In default, you don't have the `order_id` to determine the applied discount.